pkg/engine/engine.go

```diff
@@ -164,11 +166,24 @@ func WithFilterEntropy(entropy float64) Option {
 	}
 }
 
-// WithOnlyVerified sets the onlyVerified flag on the engine. If set to true,
-// the engine will only print verified results.
-func WithOnlyVerified(onlyVerified bool) Option {
+// WithResults defines which results will be printed by the engine.
+func WithResults(results map[string]struct{}) Option {
 	return func(e *Engine) {
-		e.onlyVerified = onlyVerified
+		if len(results) > 0 {
+			if _, ok := results["verified"]; ok {
+				e.notifyVerifiedResults = true
+			}
+			if _, ok := results["unknown"]; ok {
+				e.notifyUnknownResults = true
+			}
+			if _, ok := results["unverified"]; ok {
+				e.notifyUnverifiedResults = true
+			}
+		} else {
+			e.notifyVerifiedResults = true
+			e.notifyUnknownResults = true
+			e.notifyUnverifiedResults = true
+		}
```

Comment:
I presume that this is implemented to mimic the previous behavior of turning options _off_, but I find it surprising to read on its own. Since we're setting the new flag default value, is it possible to dispense with effectively duplicating the default behavior here and instead implement this like a more "normal" functional option?

Reply:
To clarify: I was unexpected to see that passing an empty map to `WithResults` results in showing _all_ results, when passing a single-element map to `WithResults` only shows those results.

Reply:
An empty map would signify that nothing is specified (e.g., `trufflehog git file:///tmp/repo` without a `--results` flag.) I believe this is consistent with the current behaviour.

> Since we're setting the new flag default value, is it possible to dispense with effectively duplicating the default behavior here and instead implement this like a more "normal" functional option?

The code is admittedly awkward. I wanted to handling parsing + assignment in `WithResults`, however, `Option` does not allow returning errors. Hence, the raw input is parsed in `main.go#parseResults` and passed to `WithResults` as a map.

Off the top of my head, to moving everything into `WithResults` would require either modifying the `Option` type or manually calling `os.Exit()` inside the function.

Reply:
Ah, I'm proposing establishing the default _before_ `WithResults` is even called and then having `WithResults` directly translate the map into struct options without worrying about default behavior for empty maps. It would move the specification of the default behavior higher in the call tree and, in my opinion, make `WithResults` more straightforward. But it's not a big deal either way!

Reply:
Oh, I see what you're saying. I'll do that.